sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -633,13 +658,96 @@ public Builder setAccessToken(String accessToken) {
     }
 
     /**
+     * Sets the client's certificate and key for TLS client authentication.
+     * @param certPath filepath to pem encoded X.509 certificate
+     * @param keyPath filepath to pem encoded X.509 private key
+     */
+    public Builder setX509KeyPair(String certPath, String keyPath) throws HTTPException {
+      try (InputStream pemStream =
+          new ByteArrayInputStream(Files.readAllBytes(Paths.get(certPath)))) {
+        // parse certificate
+        CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        X509Certificate certificate = (X509Certificate) factory.generateCertificate(pemStream);
+
+        // The PKCS standard used by the private key must be determined.
+        // Currently, Chain Core supports PKCS#1 and PKCS#8.
+        // In both cases, the header and footer ascii characters must be stripped and the remaining
+        // Base64 data decoded to DER encoded bytes. In the case of PKCS#8, the standard library
+        // provides a class representation that accepts these bytes as inputs.
+        // For PKCS#1 we must parse these bytes for specific integer values representing
+        // the ASN.1 RSAPrivateKey: https://tools.ietf.org/html/rfc3447#appendix-A.1.2
+        byte[] keyBytes = Files.readAllBytes(Paths.get(keyPath));
```

Comment:
I think I might be in favor of adding a dependency lib that can take a vanilla PEM-encoded cert and just Do The Right Thing as an off-the-shelf operation, rather than have a bunch of bespoke code in the SDK.